### PR TITLE
[Postgres] Implementing Redshift style merge

### PR DESCRIPTION
**Pull request by Tang8330**
> [!NOTE]
> **Medium Risk**
> Changes core upsert/delete SQL generation for Postgres based on detected server version; incorrect join/where conditions could cause missed updates/inserts or unintended deletes.
> 
> **Overview**
> Postgres dialect merge logic now supports an opt-out of `MERGE` by generating Redshift-style `UPDATE` + `INSERT` (+ optional `DELETE`) statements, including soft-delete behavior, composite keys, additional join predicates, and TOAST column protection in updates.
> 
> Postgres store now detects the server major version at startup and automatically disables `MERGE` for PostgreSQL versions prior to 15, with expanded unit tests covering the no-`MERGE` query generation and aliasing edge cases.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 59ebac00ee243d03e7ddc6b88a0a0aa2fbb08c11. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

File: clients/postgres/dialect/dialect.go [hunk 1]
```diff
@@ -27,7 +27,13 @@ LEFT JOIN pg_catalog.pg_attribute a ON a.attname = c.column_name AND a.attrelid
 LEFT JOIN pg_catalog.pg_attrdef d ON d.adrelid = a.attrelid AND d.adnum = a.attnum
 WHERE c.table_schema = $1 AND c.table_name = $2;`
 
-type PostgresDialect struct{}
+type PostgresDialect struct {
+	disableMerge bool
+}
+
+func NewPostgresDialect(disableMerge bool) PostgresDialect {
+	return PostgresDialect{disableMerge: disableMerge}
```
[the remark this answers is not in the document]
Comment (Tang8330) [reply clients/postgres/dialect/dialect.go]: Yeah, I'll build it based on that. Thanks for the suggestion!

File: clients/postgres/dialect/dialect.go [hunk 2]
```diff
@@ -201,6 +210,109 @@ WHEN NOT MATCHED AND COALESCE(%s, false) = false THEN INSERT (%s) VALUES (%s)`,
...
+	tableID sql.TableIdentifier,
+	subQuery string,
+	primaryKeys []columns.Column,
+	additionalEqualityStrings []string,
+	cols []columns.Column,
+	softDelete bool,
+	containsHardDeletes bool,
+) ([]string, error) {
+	if !softDelete {
+		cols, err := columns.RemoveDeleteColumnMarker(cols)
```
Comment (AnirudhSriram) [inline clients/postgres/dialect/dialect.go]: might want to declare err before ? cols parameter shadowing here.

**Review by AnirudhSriram — APPROVED** [review]
